### DISCUSSION TITLE
[General] Error : Invalid EVM version requested using foundry-zksync and in-memory era_test_node

### DISCUSSION THREAD

**Discussion by Alessandro-Cavaliere** [zkSync-Community-Hub/zksync-developers#641] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Hi all, 

I'm trying to deploy to my local **ZKsync** node the**SimpleStorage.sol** contract, but to no avail.
The command I'm running is the following:

`forge create ./src/SimpleStorage.sol:SimpleStorage --rpc-url http://127.0.0.1:8011 --private-key 0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110 --legacy --zksync`

I used all the mandatory flags in the documentation and pasted in one of the **private keys** that the documentation provides, I entered the correct **rpc-url**, my **docker** is up and i'm using the right forge version with **foundry-zksync**, but it returns this **output**:

> [⠃] Using zksolc-1.5.1
> [⠊] Compiling (zksync)
> [⠒] Compiling 1 files with ZkSolc 0.8.19
> [⠢] ZkSolc 0.8.19 finished in 5.51ms
> Error: 
> Compiler run failed:
> Error
> **Invalid EVM version requested.**

It appears to be using all the **ZkSocl** compilers as the documentation indicates, but it's as if the `--zksync` command didn't fire at all as it doesn't recognize the **zkEVM**.

If anyone has already encountered this error, please let me know if it has been resolved somehow.

**Comment by alhenaai** [comment]
It seems like you are encountering an issue with the `--zksync` flag not properly initiating the zkEVM environment, which is causing the `Invalid EVM version requested` error. This could be due to a mismatch or an unsupported version of the ZkSolc compiler.

Here are a few steps you can try to resolve this issue:

1. **Verify ZkSolc Version**: Ensure that the version of ZkSolc you are using is compatible with the ZKsync environment. You can specify a different version using the `--use-zksolc <ZK_SOLC_VERSION>` option in your command.

2. **Update Foundry-zksync**: Make sure that your foundry-zksync tool is up to date. You can update it using the command:
   ```
   cargo install foundry-zksync --force
   ```

3. **Check Compiler Configuration**: Review the compiler settings in your foundry configuration to ensure they are set correctly for ZKsync. You might need to adjust settings like optimization levels or system modes.

4. **Consult Documentation**: Double-check the [ZKsync documentation](https://docs.zksync.io/build/tooling/foundry/getting-started#compilation-with-forge-build-zksync) for any updates or additional flags that might be required for your specific setup.

If these steps do not resolve the issue, it might be helpful to reach out on the official forums or support channels for more specific guidance tailored to your setup.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by elfedy** [comment, marked as answer]
@Alessandro-Cavaliere the error comes from solidity due to the evm version being passed as input. Foundry's default evm version is cancun but the solc version it is trying to compile with is 0.8.19 which is not compatible with `cancun`

so either try bumping the version in the solidity contracts to `0.8.24` or specify the evm version on foundry.toml like this:
```
[profile.default]
...
evm_version = "paris"
```
In the near future we will santize the input so that the highest compatible evm version for each solc is used but for now that workaround is needed
